### PR TITLE
Trim unused proto.

### PR DESCRIPTION
**Pull request by jkh52**
Trim unused proto.

### CODE REVIEW

**Comment by tallclair** [comment]
/hold
for a later release

/assign @cheftako

**Comment by cheftako** [comment]
You don't seem to have regenerated konnectivity-client/proto/client/client.pb.go. I think you at minimum need to do that and see what the consequences of those changes are.

**Comment by jkh52** [comment]
> You don't seem to have regenerated konnectivity-client/proto/client/client.pb.go.

whoops - fixed thanks.

**Comment by jkh52** [comment]
@cheftako I rebased this PR after #434, but there appears to be fallout:

```
protoc -I . proto/agent/agent.proto --go_out=. --go_opt=paths=source_relative --go-grpc_out=require_unimplemented_servers=false:. --go-grpc_opt=paths=source_relative
make[1]: protoc: Command not found
make[1]: *** [Makefile:116: proto/agent/agent.pb.go] Error 127
make[1]: Leaving directory '/home/prow/go/src/sigs.k8s.io/apiserver-network-proxy'
```

File: Makefile [hunk 1]
```diff
@@ -50,6 +50,7 @@ PROXY_SERVER_IP ?= 127.0.0.1
 ## --------------------------------------
 mock_gen:
 	mkdir -p proto/agent/mocks
+	rm -f proto/agent/mocks/agent_mock.licensed.go proto/agent/agent_grpc.licensed.go
```
Comment (cheftako) [inline Makefile:53]: I don't think we should need this line as a successful make should always move the file, hence cleaning it up. So I happy to have this in clean but I don't think it should be part of the normal make. The fact that we don't have client/proto/client/client_grpc.licensed.go here supports that.

If we do decide to keep it I would rather have this in the phony top level target can we move the deletes to the proper file targets? (Eg. `proto/agent/agent_grpc.pb.go proto/agent/agent.pb.go:`)

Comment (jkh52) [reply Makefile]: You're right that it seems to not belong here.

I removed it and re-tested `make gen` a few times, at first it began mysteriously succeeding (where it failed before, with "unexpected EOF"). Then on the most recent try, it seemed to succeed but the resulting proto/agent/agent_grpc.pb.go was missing the license header. So, I think as-is there is some non-determinism.

**Pull request merged**